core/src/main/java/feast/core/model/FeatureStatistics.java

```diff
@@ -0,0 +1,269 @@
+    })
+public class FeatureStatistics {
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private int id;
+
+  @ManyToOne
+  @JoinColumns({
+    @JoinColumn(name = "project", referencedColumnName = "project"),
+    @JoinColumn(name = "feature_set", referencedColumnName = "feature_set"),
+    @JoinColumn(name = "version", referencedColumnName = "version"),
+    @JoinColumn(name = "name", referencedColumnName = "name")
+  })
+  private Feature feature;
+
+  // Only one of these fields should be populated.
+  private String datasetId;
+  private Date date;
+
+  // General statistics
+  private String featureType;
+  private long count;
+  private long numMissing;
+  private long minNumValues;
+  private long maxNumValues;
+  private float avgNumValues;
+  private long totalNumValues;
+  private byte[] numValuesHistogram;
+
+  // Numeric statistics
+  private double mean;
+  private double stdev;
+  private long zeroes;
+  private double min;
+  private double max;
+  private double median;
+  private byte[] numericValueHistogram;
+  private byte[] numericValueQuantiles;
+
+  // String statistics
+  @Column(name = "n_unique")
+  private long unique;
+
+  private float averageLength;
+  private byte[] rankHistogram;
+  private byte[] topValues;
+
+  // Byte statistics
+  private float minBytes;
+  private float maxBytes;
+  private float avgBytes;
+
+  // Instantiates a Statistics object from a tensorflow metadata FeatureNameStatistics object and a
+  // dataset ID.
+  public static FeatureStatistics fromProto(
+      String project,
+      String featureSetName,
+      int version,
+      FeatureNameStatistics featureNameStatistics,
+      String datasetId)
+      throws IOException {
+    FeatureStatistics featureStatistics = FeatureStatistics.fromProto(featureNameStatistics);
+    Feature feature =
+        Feature.withReference(
+            new FeatureReference(
+                project, featureSetName, version, featureNameStatistics.getPath().getStep(0)));
+    featureStatistics.setFeature(feature);
+    featureStatistics.setDatasetId(datasetId);
+    return featureStatistics;
+  }
+
+  // Instantiates a Statistics object from a tensorflow metadata FeatureNameStatistics object and a
+  // date.
+  public static FeatureStatistics fromProto(
+      String project,
+      String featureSetName,
+      int version,
+      FeatureNameStatistics featureNameStatistics,
+      Date date)
+      throws IOException {
+    FeatureStatistics featureStatistics = FeatureStatistics.fromProto(featureNameStatistics);
+    featureStatistics.setDate(date);
+    Feature feature =
+        Feature.withReference(
+            new FeatureReference(
+                project, featureSetName, version, featureNameStatistics.getPath().getStep(0)));
+    featureStatistics.setFeature(feature);
+    return featureStatistics;
+  }
+
+  public FeatureNameStatistics toProto() throws InvalidProtocolBufferException {
```

Comment:
Same as the other comments. I think it should be renamed to indicate that this is exporting a TensorFlow `FeatureNameStatistics` object.